### PR TITLE
gwcli: add reconfigure command to tune various lun attributes

### PR DESCRIPTION
**Pull request by vshankar**
This command allows the user to modify and/or reset to default
various LUN attributes. To start with, this allows to modify
(or reset) the ring buffer size (max_data_area_mb).

Command format:

    - modify max_data_area_mb
      reconfigure image_id=rbd.disk_1 max_data_area_mb=128

    - reset max_data_area_mb
      reconfigure image_id=rbd.disk_1 max_data_area_mb=

Signed-off-by: Venky Shankar <vshankar@redhat.com>

### CODE REVIEW

**Comment by vshankar** [comment]
Note that the backend (API) still accepts multiple attributes to reconfigure, but the CLI accepts (and passes) one at a time

File: gwcli/storage.py [hunk 1]
```diff
@@ -750,6 +805,9 @@ def ui_command_resize(self, size=None):
 
         self.resize(size)
 
+    def ui_command_reconfigure(self, attribute=None, value=''):
+        self.reconfigure(attribute, value)
```
Comment (dillaman) [inline gwcli/storage.py]: I can run "reconfigure" here and it doesn't throw an error for not specifying an attribute / value:

> /disks/rbd.lun0> reconfigure 
> Issuing reconfigure request: attribute=None, value=
> ok
>

File: gwcli/storage.py [hunk 2]
```diff
@@ -378,6 +378,37 @@ def ui_command_resize(self, image_id=None, size=None):
...
+        attribute : attribute to reconfigure
+        value     : value of the attribute to reconfigure
+
+        e.g.
+        set max_data_area_mb
+          - reconfigure rbd.disk_1 max_data_area_mb 128
+        reset max_data_area_mb to default
+          - reconfigure rbd.disk_1 max_data_area_mb
+        """
+        allowed_attributes = ['max_data_area_mb']
```
Comment (dillaman) [inline gwcli/storage.py]: Move this validation to ```reconfigure``` method below

File: gwcli/storage.py (hunk 1 above)
Comment (dillaman) [inline gwcli/storage.py]: Nit: add documentation for the attributes

File: gwcli/storage.py [hunk 3]
```diff
@@ -750,6 +805,9 @@ def ui_command_resize(self, size=None):
 
         self.resize(size)
 
+    def ui_command_reconfigure(self, attribute=None, value=''):
```
Comment (dillaman) [inline gwcli/storage.py]: Nit: don't default the parameters so that it forces the user to specify them

File: gwcli/storage.py [hunk 4]
```diff
@@ -378,6 +378,37 @@ def ui_command_resize(self, image_id=None, size=None):
             self.logger.error("resize needs the disk image name and new size")
             return
 
+    def ui_command_reconfigure(self, image_id=None, attribute=None, value=''):
```
Comment (dillaman) [inline gwcli/storage.py]: Nit: don't default the parameters so that it forces the user to specify them

File: gwcli/storage.py (hunk 1 above)
Comment (vshankar) [reply gwcli/storage.py]: Thanks for testing this. I moved the validations to `Disks::reconfigure` method.

File: gwcli/storage.py [hunk 5]
```diff
@@ -750,6 +803,9 @@ def ui_command_resize(self, size=None):
 
         self.resize(size)
 
+    def ui_command_reconfigure(self, attribute, value):
+        self.reconfigure(attribute, value)
```
Comment (dillaman) [inline gwcli/storage.py]: ... still need to add the documentation (derived from the ```ui_command_reconfigure``` above)

Comment (vshankar) [reply gwcli/storage.py]: done

**Pull request merged**